### ISSUE TITLE
Rule proposal: `prefer-destructuring-in-parameters`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#954]
I think it's nicer, especially in simple functions.

## Fail

```js
foo.map(element => element.id);
```

```js
Object.entries(foo).filter(pair => pair[1]);
```

## Pass

```js
foo.map(({id}) => id);
```

```js
Object.entries(foo).filter(([, value])=> value);
```

WDYT?

**Comment by sindresorhus** [comment]
This should fail though:

```js
Object.entries(foo).filter(([,, value])=> value);
```

**Comment by fisker** [comment]
Yes, it should not against [`no-unreadable-array-destructuring`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-unreadable-array-destructuring.md)

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
Let's close this too, it reduces readability.
